jdaviz/configs/imviz/plugins/coords_info/coords_info.py

```diff
@@ -49,41 +85,273 @@ def marks(self):
                 viewer.figure.marks = viewer.figure.marks + [self._marks[id]]
         return self._marks
 
+    def as_text(self):
+        return (f"{self.row1a_title} {self.row1a_text} {self.row1b_title} {self.row1b_text}".strip(),  # noqa
+                f"{self.row2_title} {self.row2_text}".strip(),
+                f"{self.row3_title} {self.row3_text}".strip())
+
     def reset_coords_display(self):
-        self.pixel_prefix = "Pixel"
-        self.world_label_prefix = '\u00A0'
-        self.world_label_prefix_2 = '\u00A0'
-        self.world_label_icrs = '\u00A0'
-        self.world_label_deg = '\u00A0'
-        self.world_ra = ''
-        self.world_dec = ''
-        self.world_ra_deg = ''
-        self.world_dec_deg = ''
-        self.unreliable_world = False
-        self.unreliable_pixel = False
-
-    def set_coords(self, sky, unreliable_world=False, unreliable_pixel=False):
-        celestial_coordinates = sky.to_string('hmsdms', precision=4, pad=True).split()
-        celestial_coordinates_deg = sky.to_string('decimal', precision=10, pad=True).split()
-        world_ra = celestial_coordinates[0]
-        world_dec = celestial_coordinates[1]
-        world_ra_deg = celestial_coordinates_deg[0]
-        world_dec_deg = celestial_coordinates_deg[1]
-
-        if "nan" in (world_ra, world_dec, world_ra_deg, world_dec_deg):
-            self.reset_coords_display()
-        else:
-            self.pixel_prefix = 'Pixel'
-            self.world_label_prefix = 'World'
-            self.world_label_icrs = '(ICRS)'
-            self.world_label_deg = '(deg)'
-            self.world_ra = world_ra
-            self.world_dec = world_dec
-            self.world_ra_deg = world_ra_deg
-            self.world_dec_deg = world_dec_deg
-            self.unreliable_world = unreliable_world
-            self.unreliable_pixel = unreliable_pixel
-            if unreliable_world:
-                self.world_label_prefix_2 = '(est.)'
+        self.row1a_title = '\u00A0'  # to force empty line if no other content
+        self.row1a_text = ""
+        self.row1b_title = ""
+        self.row1b_text = ""
+        self.row1_unreliable = False
+
+        self.row2_title = '\u00A0'
+        self.row2_text = ""
+        self.row2_unreliable = False
+
+        self.row3_title = '\u00A0'
+        self.row3_text = ""
+        self.row3_unreliable = False
+
+        self.icon = ""
+
+    def _viewer_mouse_clear_event(self, viewer, data=None):
+        self.reset_coords_display()
+        marks = self.marks.get(viewer._reference_id)
+        if marks is not None:
+            marks.visible = False
+
+    def _viewer_mouse_event(self, viewer, data):
+        if data['event'] in ('mouseleave', 'mouseenter'):
+            return self._viewer_mouse_clear_event(viewer, data)
```

Comment:
what's the reasoning?  I always prefer passing along the return in cases like this in case `_viewer_mouse_clear_event` is ever changed to return something.

Reply:
But the return behavior isn't consistent across the function in different logic routes. Like you say, if that hidden method starts to return something, behavior will be undefined. And its return isn't really used anywhere right now. I find that confusing.

Reply:
I still prefer this, but it's a matter of taste and philosophy.  I'll change it though and ultimately it won't matter unless we ever have either give a meaningful return, which is unlikely.